File: check_sorted.py
# Time complexity here is O(n) and auxilary space is O(1)
def check_sorted(arr):
    n=len(arr)
    for i in range(0,n):
        if arr[i]>arr[i-1]:
            return False
        return True

print(check_sorted([8,12,15]))

File: check_sorted.cpp
#include <iostream>
using namespace std;

bool check_sorted(int arr[],int n)
{
    bool flag = true;
    int res = 0;
    for (int i=1;i < n;i++){
        if (arr[i] > arr[res])
            res = i;
        else{
            flag = false;
            break;
        }
            
    }
    return flag;
}

int main()
{
    int arr[] = {2,5,9,2};
    cout << check_sorted(arr,4);
    return 0;
}